### PR TITLE
John conroy/home search to react

### LINKED ISSUE #171: unpkg is unresponsive: need to host our own packages
The request for portal-search returns "Rate exceeded."

### PR DESCRIPTION
**Pull request by john-conroy**
Create components to mimic the existing Home/Search templates to get us to a baseline of everything except for error pages inside our react app. Also moves dependencies from unpkg to node modules and fixes #171 and #198 .

### CODE REVIEW

File: context/app/static/js/Routes.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import React from 'react';
+import NoticeAlert from './components/NoticeAlert';
+import Details from './components/Details';
+import Home from './components/Home';
+import Search from './components/Search/Search';
+
+function Routes(props) {
+  const { flaskData } = props;
+  const {
+    flashed_messages, entity, provenance, vitessce_conf,
```
Comment (mccalluc) [inline context/app/static/js/Routes.jsx:11]: I'd say these should be renamed on assignment to be consistent with JS naming conventions? But not 100% sure it's worth the trouble, either. Thoughts?

File: context/app/static/js/Routes.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  } = flaskData;
+  const urlPath = window.location.pathname;
+
+  if (urlPath.indexOf('/browse') > -1) {
```
Comment (mccalluc) [inline context/app/static/js/Routes.jsx:15]:
```suggestion
  if (urlPath.startsWith('/browse')) {
```

File: context/app/static/js/Routes.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    );
+  }
+
+  if (urlPath === '/') {
+    return (
+      <Home />
+    );
+  }
+
+  if (urlPath.indexOf('/search') > -1) {
```
Comment (mccalluc) [inline context/app/static/js/Routes.jsx:34]:
```suggestion
  if (urlPath === '/search') {
```

File: context/app/static/js/index.jsx [hunk 4]
```diff
@@ -9,11 +9,18 @@ import App from './App';
 import Footer from './Footer';
 import Header from './Header';
 
-if (window.location.href.search('/browse/') > -1) {
+const urlPath = window.location.pathname;
+
+if (urlPath.indexOf('/browse') > -1) {
```
Comment (mccalluc) [inline context/app/static/js/index.jsx:14]: Seems to overlap with Routes.jsx -- Are they both necessary?

Comment (john-conroy) [reply context/app/static/js/index.jsx:14]: Both are necessary so long as there are situations, now just the error pages, where we need to only attach the header/footer and not any content.

File: context/app/static/js/Routes.jsx (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/Routes.jsx]: I'll create an issue and revisit.

**Comment by john-conroy** [comment]
The `core_props` variable is repeated throughout `routes_main.py` which isn't ideal, but if it's fine for now I can open an issue and revisit once we're past the demo.

File: context/app/routes_main.py [hunk 5]
```diff
@@ -31,7 +31,8 @@ def _get_client():
 
 @blueprint.route('/')
 def index():
-    return render_template('pages/index.html', types=types)
+    core_props = {'endpoints': {'esEndpoint': current_app.config['ELASTICSEARCH_ENDPOINT']}}
```
Comment (mccalluc) [inline context/app/routes_main.py:34]: Prefer to spell things out, and generally keep names consistent.
```suggestion
    core_props = {'endpoints': {'elasticsearchEndpoint': current_app.config['ELASTICSEARCH_ENDPOINT']}}
```

File: context/app/routes_main.py [hunk 6]
```diff
@@ -65,17 +66,19 @@ def details(type, uuid):
...
-    }
+    })
     return render_template(
-        template, type=type, uuid=uuid,
-        title_text='TODO: title_text',
-        flask_data=props
+        template,
+        type=type,
+        uuid=uuid,
+        flask_data=core_props
```
Comment (mccalluc) [inline context/app/routes_main.py:81]: `core_props` could be moved out to the top level, and we don't need to mutate it.
```suggestion
        flask_data={
            'flashed_messages': flashed_messages,
            'entity': entity,
            'provenance': provenance,
            'vitessce_conf': client.get_vitessce_conf(),
            **core_props
        }
```

File: context/app/routes_main.py [hunk 7]
```diff
@@ -95,10 +98,11 @@ def details_ext(type, uuid, ext):
 
 @blueprint.route('/search')
 def search():
+    core_props = {'endpoints': {'esEndpoint': current_app.config['ELASTICSEARCH_ENDPOINT']}}
```
Comment (mccalluc) [inline context/app/routes_main.py:101]:
```suggestion
```
Move to top level

File: context/app/routes_main.py (hunk 5 above)
Comment (john-conroy) [reply context/app/routes_main.py:34]: I initially did that but I was getting a flask error similar to `RuntimeError: working outside of application context` .

File: context/app/static/js/Routes.jsx (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/Routes.jsx]: Created #258

**Pull request merged**